modules/nlu/src/backend/engine.ts

```diff
@@ -375,18 +378,20 @@ export default class ScopedEngine implements Engine {
   private _extractIntents = async (ds: NLUStructure): Promise<NLUStructure> => {
     const exactMatcher = this._exactIntentMatchers[ds.language]
     const exactIntent = exactMatcher && exactMatcher.exactMatch(ds.sanitizedText, ds.includedContexts)
-
-    const skipIntentExtraction = !((await this.getIntents()) || []).length
-    if (skipIntentExtraction) {
-      return ds
-    }
-
     if (exactIntent) {
       ds.intent = exactIntent
       ds.intents = [exactIntent]
       return ds
     }
 
+    const allIntents = (await this.getIntents()) || []
+    const isThereEnoughUtterances = !!allIntents.filter(i => i.utterances[ds.language].length >= MIN_NB_UTTERANCES)
```

Comment:
just for your info, i think this could be done with a `.some` which ensures at lease one of the results computed as a bool is truthy. And since it defaults to false you wouldn't need to check that allIntents has some values

```js
const shouldPredict = allIntents.some( i.utterances[ds.language].length >= MIN_NB_UTTERANCES))

if(!shouldPredict) {
  return ds
}
```

Reply:
Did the fix! Yes, I tried multiple scenarios.